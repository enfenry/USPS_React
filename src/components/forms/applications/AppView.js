import React from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';

const AppView = props => {
    const { handleHide, name, initialValues } = props;

    return (
        <div>
            <div>
                <span>Name: </span>
                <p>{name}</p>
            </div>
            <div>
                <span>Application Type: </span>
                <p>{initialValues['ss_applicationtype@OData.Community.Display.V1.FormattedValue']}</p>
            </div>
            <div>
                <span>Product: </span>
                <p>{initialValues['_ss_product_value@OData.Community.Display.V1.FormattedValue']}</p>
            </div>
            {initialValues['ss_applicationtype@OData.Community.Display.V1.FormattedValue'] === 'Package Submission' ?
                <div>
                    <span>Shipping Speed: </span>
                    <p>{initialValues['_ss_shippingspeed_value@OData.Community.Display.V1.FormattedValue']}</p>
                </div>
                : ""
            }
            <div>
                <span>Customer: </span>
                <p>{initialValues['_ss_customer_value@OData.Community.Display.V1.FormattedValue']}</p>
            </div>
            <div>
                <span>Destination Address: </span>
                <p>{initialValues['_ss_destinationaddress_value@OData.Community.Display.V1.FormattedValue']}</p>
            </div>
            <div className="control">
                <Button color="secondary" onClick={handleHide}>Cancel</Button>
            </div>
        </div>
    )
}

AppView.propTypes = {
    handleHide: PropTypes.func,
    name: PropTypes.string,
    initialValues: PropTypes.object,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ])
}

export default AppView;