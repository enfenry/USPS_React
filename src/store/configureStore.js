import { createStore, applyMiddleware, compose } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

import { createLogger } from 'redux-logger'
import DevTools from '../containers/DevTools';

function configureStore(initialState) {

  const logger = createLogger()

  const middlewares = [
    reduxImmutableStateInvariant(),
    thunk,
    logger
  ];

  const store = createStore(
    rootReducer,
    initialState,
    compose(
      applyMiddleware(...middlewares),
      DevTools.instrument()
    )
  );

  return store;
}

export default configureStore;
