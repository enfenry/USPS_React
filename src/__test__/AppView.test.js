import React from 'react';
import AppView from '../components/forms/applications/AppView';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import * as basicStateUtils from "./basicState";

import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);

test('AppView renders correctly', () => {

    const state = basicStateUtils.getBasicState();
    const store = mockStore(state);
    const initialValues = state.applicationsReducer.applications[0];

    const component = renderer.create(
        <Provider store={store}>
            <AppView
                name="example name"
                initialValues={{ ...initialValues, appTypeLabel: "example app type" }}
            />
        </Provider>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});