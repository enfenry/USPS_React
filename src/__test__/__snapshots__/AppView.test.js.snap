// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView renders correctly 1`] = `
<div>
  <div>
    <div>
      <span>
        Name: 
      </span>
      <p>
        example name
      </p>
    </div>
  </div>
  <div>
    <div>
      <span>
        Application Type: 
      </span>
      <p>
        example app type
      </p>
    </div>
  </div>
  <div>
    <div>
      <span>
        Product: 
      </span>
      <p>
        1 Year
      </p>
    </div>
  </div>
  
  <div>
    <div>
      <span>
        Customer: 
      </span>
      <p>
        Mojo Jojo
      </p>
    </div>
  </div>
  <div>
    <div>
      <span>
        Destination Address: 
      </span>
      <p>
        Work Apartments
      </p>
    </div>
  </div>
  <div
    className="control"
  >
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;
